lib/Sema/TypeCheckProtocolInference.cpp

```diff
@@ -188,22 +188,27 @@ AssociatedTypeInference::inferTypeWitnessesViaValueWitnesses(
     // Invalid case.
     if (extendedNominal == nullptr)
       return true;
-
+    
+    // Validate the nominal type being extended.
+    tc.validateDecl(extendedNominal);
```

Comment:
So validating a nominal doesn't do a whole lot anymore. It just calls computeType() which is trivial. Can you humor me and try removing some of these calls?